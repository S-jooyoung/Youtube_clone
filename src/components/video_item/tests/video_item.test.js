//import dependencies
import React from "react";

//import custom-component
import VideoItem from "../video_item";

//import snapshot-testing methods
import renderer from "react-test-renderer";

describe("VideoItem", () => {
  const video = {
    id: "cheYIVEtVQ4",
    snippet: {
      title: "Test Title",
      description: "Test Desciption",
      channelTitle: "TestCode is good",
      defaultAudioLanguage: "en",
      thumbnails: {
        default: {
          url: "https://i.ytimg.com/vi/cheYIVEtVQ4/default.jpg",
          width: 120,
          height: 90,
        },
        medium: {
          url: "https://i.ytimg.com/vi/cheYIVEtVQ4/mqdefault.jpg",
          width: 320,
          height: 180,
        },
        high: {
          url: "https://i.ytimg.com/vi/cheYIVEtVQ4/hqdefault.jpg",
          width: 480,
          height: 360,
        },
        standard: {
          url: "https://i.ytimg.com/vi/cheYIVEtVQ4/sddefault.jpg",
          width: 640,
          height: 480,
        },
        maxres: {
          url: "https://i.ytimg.com/vi/cheYIVEtVQ4/maxresdefault.jpg",
          width: 1280,
          height: 720,
        },
      },
    },
  };
  const display = "list";

  it("renderer", () => {
    const component = renderer
      .create(
        <VideoItem video={video} onVideoClick={jest.fn()} display={display} />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
