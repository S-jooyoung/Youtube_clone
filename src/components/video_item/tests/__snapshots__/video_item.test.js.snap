// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem renderer 1`] = `
<li
  className="container list"
  data-testid="video_item"
  onClick={[Function]}
  title="videoItem"
>
  <div
    className="video"
  >
    <img
      alt="video thumbnail"
      className="thumbnail"
      src="https://i.ytimg.com/vi/cheYIVEtVQ4/mqdefault.jpg"
    />
    <div
      className="metadata"
    >
      <p
        className="title"
        data-testid="title"
      >
        Test Title
      </p>
      <p
        className="channel"
        data-testid="channel"
      >
        TestCode is good
      </p>
    </div>
  </div>
</li>
`;
